packages/core/src/components/icon/icon.tsx

```diff
@@ -17,50 +17,40 @@ export interface IIconProps extends IIntentProps, IProps {
      * Name of the icon (with or without `"pt-icon-"` prefix).
      * If `undefined`, this component will render nothing.
      */
-    iconName: IconName | undefined;
+    iconName: LegacyIconName | undefined;
 
     /**
-     * Size of the icon.
-     * Blueprint provides each icon in two sizes: 16px and 20px. The keyword `"inherit"` will
-     * render a 20px icon but inherit `font-size` from its parent.
-     * Constants are exposed for each of these values on the component itself:
-     * `Icon.SIZE_(STANDARD|LARGE|INHERIT)`,
+     * Height of icon. A number value will be interpreted as pixels. Use a string value for other units.
+     * By default, inherits height from surrounding styles, such as `line-height`.
+     * @default "inherit"
+     */
+    height?: number | string;
+
+    /**
+     * Width of icon. A number value will be interpreted as pixels. Use a string value for other units.
      * @default 16
      */
-    iconSize?: 16 | 20 | "inherit";
+    width?: number | string;
 }
 
 export class Icon extends React.PureComponent<IIconProps & React.HTMLAttributes<HTMLSpanElement>, never> {
     public static displayName = "Blueprint.Icon";
 
     public static readonly SIZE_STANDARD = 16 as 16;
     public static readonly SIZE_LARGE = 20 as 20;
-    public static readonly SIZE_INHERIT = "inherit" as "inherit";
 
     public render() {
-        if (this.props.iconName == null) {
+        const { className, iconName, intent, width = 16, height = "inherit" } = this.props;
+        if (iconName == null) {
             return null;
         }
-        const { className, iconName, intent, iconSize = Icon.SIZE_STANDARD, ...restProps } = this.props;
-
-        const classes = classNames(
-            getSizeClass(iconSize),
-            Classes.iconClass(iconName),
-            Classes.intentClass(intent),
-            className,
+        const shortName = iconName.replace("pt-icon-", "") as IconName;
+        const classes = classNames(Classes.ICON, Classes.iconClass(iconName), Classes.intentClass(intent), className);
+        return (
+            <svg className={classes} width={width} height={height} viewBox="0 0 16 16">
```

Comment:
shouldn't `viewBox` use the size/width props?

Reply:
no, viewBox is based on the size of the image itself, so it's either 16 or 20. width/height scale the viewBox.